OSDOCS-3009:Adding route compression to Ingress (#39982)
openshift/openshift-docs: modules/nw-ingress-controller-configuration-parameters.adoc

For request headers, these adjustments are applied only for routes that have the `haproxy.router.openshift.io/h1-adjust-case=true` annotation. For response headers, these adjustments are applied to all HTTP responses. If this field is empty, no request headers are adjusted.

|`httpCompression`

> |`httpCompression` defines the policy for HTTP traffic compression. The optional `httpCompressionPolicy` field turns on compression for the MIME types, which are specified in the `RouterCompressionMIMETypes` variable. Compression is not enabled by default.

Reviewer: `The optional `httpCompressionPolicy` `, i think 'The optional `ROUTER_ENABLE_COMPRESSION`

You can configure compression of MIME types using the environment variables:

> * `RouterEnableCompression` if set to `true` or `TRUE`, then the body of the response is compressed when possible.

Reviewer: `RouterEnableCompression` can use 'ROUTER_ENABLE_COMPRESSION'

> * `RouterCompressionMIMETypes` defines a list of MIME types to which compression is applied.  The types are: application, image, message, multipart, text, video, or a custom type prefaced by "X-". To specify a format, use the type/subtype [;attribute=value]. For example, "text/css; charset=utf-8", "text/html", "text/*", "image/svg+xml", "application/octet-stream", "X-custom/customsub".

Reviewer: `RouterCompressionMIMETypes` can use 'mimeTypes'

For request headers, these adjustments are applied only for routes that have the `haproxy.router.openshift.io/h1-adjust-case=true` annotation. For response headers, these adjustments are applied to all HTTP responses. If this field is empty, no request headers are adjusted.

|`httpCompression`

> |`httpCompression` defines the policy for HTTP traffic compression. The optional `ROUTER_ENABLE_COMPRESSION` field turns on compression for the MIME types, which are specified in the `mimeTypes` variable. Compression is not enabled by default.

Reviewer 2: There is no other field except mimeTypes under httpCompression.
```suggestion
|`httpCompression` defines the policy for HTTP traffic compression. 
```

> You can configure compression of MIME types using the environment variables:

Reviewer 2: ```suggestion
```

> * `ROUTER_ENABLE_COMPRESSION` compresses the body of the response when set to `true` or `TRUE` whenever possible.

Reviewer 2: ```suggestion

```

> * `ROUTER_ENABLE_COMPRESSION` if set to `true` or `TRUE`, then the body of the response is compressed when possible.

Reviewer 3: I think this should flow as a sentence or a colon should be put after the variable to follow similar examples in the same module. For example:

`ROUTER_ENABLE_COMPRESSION`: If set to true. . .

Alternatively:

`ROUTER_ENABLE_COMPRESSION`compresses the body of the response when set to `true` or `TRUE` when possible.

> * `mimeTypes` defines a list of MIME types to which compression is applied.  The types are: application, image, message, multipart, text, video, or a custom type prefaced by `X-`. To specify a format, use the type/subtype [;attribute=value]. For example, `text/css; charset=utf-8`, `text/html`, `text/*`, `image/svg+xml`, `application/octet-stream`, `X-custom/customsub`.

Reviewer 2: ```suggestion
* `mimeTypes` defines a list of MIME types to which compression should be applied.  For example, `text/css; charset=utf-8`, `text/html`, `text/*`, `image/svg+xml`, `application/octet-stream`, etc, where format pattern `type/subtype; [;attribute=value]` is used.  The `types` are: application, image, message, multipart, text, video, or a custom type prefaced by `X-`; e.g. `X-custom/customsub`.
```

> 

Reviewer 2: There is a useful `note` in https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L257-L261

It also might not be a bad idea to reference the required formatting of a single mimeType, see https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L271-L284

Author: Ok I added the same link from the Configuring router compression topic.

> * `mimeTypes` defines a list of MIME types to which compression is applied.  The types are: application, image, message, multipart, text, video, or a custom type prefaced by "X-". To specify a format, use the type/subtype [;attribute=value]. For example, "text/css; charset=utf-8", "text/html", "text/*", "image/svg+xml", "application/octet-stream", "X-custom/customsub".

Reviewer 3: Wasn't sure about this but I'm going to message you about it.
